module/Database/src/Model/AuditNote.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Model;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+
+/**
+ * Class for registering manual notes
+ */
+#[Entity]
+class AuditNote extends AuditEntry
+{
+    /** @psalm-suppress InvalidClassConstantType */
+    protected bool $IMMUTABLE = false;
+
+    /**
+     * The note itself
+     */
+    #[Column(type: 'string')]
+    protected ?string $note = null;
+
+    public function getNote(): ?string
+    {
+        return $this->note;
+    }
+
+    public function setNote(string $note): void
+    {
+        $this->note = $note;
+    }
+
+    /**
+     * Get a textual representation of this audit entry
+     */
+    protected function getStringBodyFormatted(): string
+    {
+        return '<strong>Note</strong> on <emph>%s</emph>: <br/>%s';
+    }
+
+    /**
+     * @return array<?string>
+     */
```

Comment:
I assume the potential `null` in this array comes from the fact that `$member` is nullable in `AuditEntry`. In that case I would prefer `?->` to be used to clearly show that it can be `null`.

Since `AuditNote`s are (currently) only applied to `Member`s we may as well

```php
#[Entity]
#[AssociationOverrides([
    new AssociationOverride(
        name: 'member',
        joinColumns: new JoinColumn(
            name: 'member',
            referencedColumnName: 'lidnr',
            onDelete: 'cascade',
            nullable: false,
        ),
    ),
])]
class AuditNote extends AuditEntry
{
    // ...


    /**
     * Get the member.
     *
     * @psalm-suppress InvalidNullableReturnType
     */
    public function getMember(): Member
    {
        return $this->member;
    }
```

to force a non-`null` return on `getMember()`.

Reply:
Changed